docs/user_guide/fairness_in_machine_learning.rst

```diff
@@ -141,6 +141,132 @@ which groups these values were observed, as well as the difference and ratio
+problems within a community, for example lack of education, infrastructure, information
+technology and health services, then an algorithmic solution purely focused on internet
+speed will fail to meaningfully address the needs of the community.
+
+
+The Ripple Effect Trap
+^^^^^^^^^^^^^^^^^^^^^^
+
+Selbst et al. [#4]_ define this as a "failure to understand how the insertion of
+technology into an existing social system changes the behaviors and embedded values
+of the pre-existing system".
+
+Example 1:
+
+The Formalism Trap
+^^^^^^^^^^^^^^^^^^
+
+This trap occurs when implementing an algorithmic solution that fails to take
+into account the social dimensions associated to fairness in a situation. These
+dimensions include "procedurality", "contextuality" and "contestability". Such
+dimensions can often not be resolved via a purely mathematical framework.
+
+Example:
+
+The Portability Trap
+^^^^^^^^^^^^^^^^^^^^
+
+This trap occurs when we fail to understand how reusing a model or
+algorithm that is designed for one specific social context, may not necessarily
+apply to a different social context. Reusing an algorithmic solution and failing
+to take into account differences in involved social contexts can result in misleading
+results and potentially harmful consequences if the algorithm is used to determine the
+fate of an individual.
+
+Example: Reusing a machine learning algorithm used to screen job applications in the
+nursing industry, for job applications in the information technology sector. An intuitive
+yet important difference between both contexts is the difference in skills required to
+succeed in both industries. A slightly more subtle difference is the demographic differences
+in gender attraction to each of the industries, resulting from wording in job postings,
+social constructs on gender and societal roles, and the male-female ratio of successful
+applicants in each field.
+
+The Framing Trap
+^^^^^^^^^^^^^^^^
+
+This trap occurs when data scientists fail to consider the full picture surrounding
+a particular social context when abstracting a social problem, and implementing an
+algorithm in which the outcome involves enforcing decisions that will impact a person
+or group of people.
+
+The authors define three main frames:
+
+1. The algorithmic frame: these are choices made when abstracting a problem
+in the form of representations (data) and labelling (outcome). In this frame
+an algorithm is evaluated based on properties of output in relation to input.
+Examples of these properties include accuracy and generalizability to data the model
+did not train on. Fairness is typically not evaluated in this frame.
+
+2. The data frame: this frame request to the content of input and output of the
+algorithm ('data') whether in the form of abstractions, or outcomes. This additional
+frame allows us to question the inherent (un)fairness present in input and output data.
```

Comment:
This might be an easy place to put in a reference or link to ways of doing this, like datasheets (either the [paper](https://arxiv.org/pdf/1803.09010) or a package for it)

Reply:
Added brief comment and reference